test/com/google/enterprise/adaptor/DocumentHandlerTest.java

```diff
@@ -402,6 +406,56 @@ public void getDocContent(Request request, Response response)
   }
 
   @Test
+  public void testContentTransform() throws Exception {
+    ContentTransformFactory contentTransformFactory =
+        new ContentTransformFactory(
+        new ArrayList<Map<String, String>>() {
+          {
+            add(new HashMap<String, String>() {
+              {
+                put("class", SampleDocumentContentTransform.class.getName());
+              }
+            });
+          }
+        });
+    mockAdaptor = new MockAdaptor() {
+      @Override
+      public void getDocContent(final Request request, final Response response)
+          throws IOException, InterruptedException {
+        response.setContentType("image/jpeg");
+        OutputStream os = response.getOutputStream();
+        os.write("some cool stuff".getBytes(Charsets.UTF_8));
+        os.close();
+      }
+    };
+    String remoteIp = ex.getRemoteAddress().getAddress().getHostAddress();
+    DocumentHandler handler = createHandlerBuilder()
+        .setAdaptor(mockAdaptor)
+        .setFullAccessHosts(new String[]{remoteIp})
+        .setContentTransformPipeline(contentTransformFactory)
+        .build();
+    handler.handle(ex);
+    assertEquals("some changed stuff", new String(ex.getResponseBytes()));
+  }
+
+  private static class SampleDocumentContentTransform
+      extends DocumentContentTransform {
+
+    public SampleDocumentContentTransform(Map<String, String> config) {
+      super(config);
+    }
+
+    @Override
+    public void write(final byte[] b) throws IOException {
+      if (contentType.equals("image/jpeg")) {
```

Comment:
Here is an idea (let me know what you think):

To ContentTransformFactory add method:
   String calculateResultingType(String ct);

To ContentTransformFactory ctor add logic to check whether a DocumentContentTransform has static method with signature:
public static String getResultingType(String ct) 

Have Factory calculateResultingType call each getResultingType in turn.
If a DocumentContentTransform does not have a getResultingType static method then consider that a no-op.

Do the figuring out of whether there is getResultingType in ctor of ContentTransformFactory, do the calling in calculateResultingType.

What do you think of this idea? (it does not ahve to be put into this CL).